helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt

```diff
@@ -0,0 +1,187 @@
+            ?.get("postgres") as? PostgresStorageConfiguration
+            ?: throw IllegalArgumentException("postgresStorage not configured.")
+
+        return DatabaseUtils.createHikariDataSource(
+            config = storageConfig,
+            applicationNameSuffix = ORTH_NAME,
+            maxPoolSize = 1
+        ).connection
+    }
+}
+
+private data class ScannedPackage(
+    val id: Identifier,
+    val provenance: Provenance,
+    val scanTime: Instant
+)
+
+private fun getScannedPackages(connection: Connection, ids: Collection<Identifier>): List<ScannedPackage> {
+    val query = """
+        SELECT 
+            s.identifier as id,
+            s.scan_result->>'provenance' as provenance,
+            (s.scan_result->'summary'->>'start_time')::timestamp as start_time
+        FROM 
+            scan_results s 
+        WHERE 
+            s.identifier = ANY(?);
+        """.trimMargin()
+
+    val resultSet = connection.prepareStatement(query).apply {
+        val array = connection.createArrayOf("VARCHAR", ids.distinct().map { it.toCoordinates() }.toTypedArray())
+        setArray(1, array)
+    }.executeQuery()
+
+    val result = mutableListOf<ScannedPackage>()
+
+    while (resultSet.next()) {
+        val id = Identifier(resultSet.getString("id"))
+        val provenance = jsonMapper.readValue(resultSet.getString("provenance"), Provenance::class.java)
+        val startTime = resultSet.getTimestamp("start_time").toInstant()
+
+        result += ScannedPackage(id, provenance, startTime)
+    }
+
+    return result.distinct()
+}
+
+private fun createAnalyzerResult(packages: Collection<ScannedPackage>) = OrtResult.EMPTY.copy(
+    analyzer = AnalyzerRun(
+        startTime = Instant.now(),
+        endTime = Instant.now(),
+        environment = Environment(),
+        config = AnalyzerConfiguration(),
+        result = AnalyzerResult(
+            projects = sortedSetOf(),
+            packages = packages.mapTo(sortedSetOf()) { CuratedPackage(it.toPackage()) }
+        )
+    )
+)
+
+private fun Collection<ScannedPackage>.filterMaxByDate(): List<ScannedPackage> {
+    fun filterMaxByDate(predicate: (ScannedPackage) -> Boolean) = filter(predicate).groupBy {
+        it.id
+    }.mapNotNull { (_, packages) -> packages.maxByOrNull { it.scanTime } }
+
+    return filterMaxByDate { it.provenance is RepositoryProvenance } +
+        filterMaxByDate { it.provenance is ArtifactProvenance }
+}
+
+private fun ScannedPackage.toPackage() = Package(
+    id = id,
+    declaredLicenses = sortedSetOf(),
+    concludedLicense = null,
+    description = "",
+    homepageUrl = "",
+    binaryArtifact = RemoteArtifact.EMPTY,
+    sourceArtifact = if (provenance is ArtifactProvenance) {
+        provenance.sourceArtifact
+    } else {
+        RemoteArtifact.EMPTY
+    },
+    vcs = if (provenance is RepositoryProvenance) {
+        provenance.vcsInfo.copy(revision = provenance.resolvedRevision)
+    } else {
+        VcsInfo.EMPTY
+    }
```

Comment:
Maybe assign `vcsProcessed` as well to make sure that normalization does not mess with the URL.